koku/api/common/permissions/settings_access.py

```diff
@@ -11,6 +12,32 @@ class SettingsAccessPermission(permissions.BasePermission):
 
     resource_type = "settings"
 
+    def has_permission(self, request, view):
+        """Check permission based on the defined access."""
+        if settings.ENHANCED_ORG_ADMIN and request.user.admin:
+            return True
+
+        if not request.user.access:
+            return False
+
+        if request.method in permissions.SAFE_METHODS:
+            setting_read = request.user.access.get(self.resource_type, {}).get("read", [])
+            if setting_read:
+                return True
+        else:
+            setting_write = request.user.access.get(self.resource_type, {}).get("write", [])
+            if "*" in setting_write:
```

Comment:
```suggestion
            if "*" in setting_write := request.user.access.get(self.resource_type, {}).get("write", []):
```

Reply:
```
    if "*" in setting_write := request.user.access.get(self.resource_type, {}).get("write", []):
       ^
SyntaxError: cannot use assignment expressions with comparison
```

Reply:
Ah, that's right. I forgot about some of the weirdness around assignment expressions. Need the wrap the expression in parenthesis.
```python
if "*" in (setting_write := request.user.access.get(self.resource_type, {}).get("write", []))
    ...
```